### PR TITLE
set up tanstack router

### PR DESCRIPTION
**Pull request by AnasChebili**

### CODE REVIEW

File: src/routes/__root.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        <Link to="/recruitment" className="[&.active]:font-bold">
+          Recruitment
+        </Link>
+        <Link to="/time-tracking" className="[&.active]:font-bold">
+          Time Tracking
+        </Link>
+      </div>
+      <hr />
+      <Outlet />
+      <TanStackRouterDevtools />
```
Comment (work-go) [inline src/routes/__root.tsx]: If we can only render this on dev, I feel like that would be amazing.

**Review by work-go — CHANGES_REQUESTED** [review]

File: src/routes/__root.tsx (hunk 1 above)
Comment (AnasChebili) [reply src/routes/__root.tsx]: I don't know what that means, would you be so kind as to spearhead the explanation of your request

Comment (muneeebnaveeed) [reply src/routes/__root.tsx]: Node (and browser) usually keep track of different environments (e.g. dev, staging, prod) under the environment variable called `NODE_ENV`. We can access this using `process.env.NODE_ENV`. Vite handles this using [modes](https://vite.dev/guide/env-and-mode#modes).

In this case, the conditional render would look something like:
`import.meta.env.MODE === "development" && <TanStackRouterDevtools />`

Comment (AnasChebili) [reply src/routes/__root.tsx]: Done!

**Review by muneeebnaveeed — APPROVED** [review]

**Review by work-go — APPROVED** [review]

**Pull request merged**